docs/readconsoleinputex.md

```diff
@@ -0,0 +1,161 @@
+---
+title: ReadConsoleInputEx function
+description: Reads data from a console input buffer with a specified set of flags and removes it from the buffer.
+author: DHowett
+ms.author: duhowett
+ms.topic: article
+keywords: console, character mode applications, command line applications, terminal applications, console api
+f1_keywords:
+- consoleapi/ReadConsoleInputEx
+- wincon/ReadConsoleInputEx
+- ReadConsoleInputEx
+- consoleapi/ReadConsoleInputExA
+- wincon/ReadConsoleInputExA
+- ReadConsoleInputExA
+- consoleapi/ReadConsoleInputExW
+- wincon/ReadConsoleInputExW
+- ReadConsoleInputExW
+MS-HAID:
+- '\_win32\_readconsoleinputex'
+- 'base.readconsoleinputex'
+- 'consoles.readconsoleinputex'
+MSHAttr:
+- 'PreferredSiteName:MSDN'
+- 'PreferredLib:/library/windows/desktop'
+ms.assetid: 5a9f7b18-cdea-4041-a377-5532d30e0105
+
+topic_type:
+- apiref
+api_name:
+- ReadConsoleInputEx
+- ReadConsoleInputExA
+- ReadConsoleInputExW
+api_location:
+- Kernel32.dll
+- API-MS-Win-Core-Console-l1-1-0.dll
+- KernelBase.dll
+- API-MS-Win-DownLevel-Kernel32-l1-1-0.dll
+- MinKernelBase.dll
+api_type:
+- DllExport
+---
+
+# ReadConsoleInputEx function
+
+Reads data from a console input buffer and removes it from the buffer, with configurable behavior.
+
+<div class="alert"><b>Note</b> This function has no associated import library. This function is available as the resources named <b>ReadConsoleInputExA</b> and <b>ReadConsoleInputExW</b> in Kernel32.dll. You must use the <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya">LoadLibrary</a> and <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-getprocaddress">GetProcAddress</a> functions to dynamically link to Kernel32.dll.</div><div> </div>
```

Comment:
I presume you followed this pattern from another doc on `docs.microsoft` that has the same problem (like `NtOpenFile` and friends.)

Reply:
Yup stole it from RtlGenCrypt or something